### DISCUSSION TITLE
Configuring the A32NX

### DISCUSSION THREAD

**Discussion by P1NBA11ER** [maartentamboer/X-Touch-Mini-FS2020#89] [Q&A]
So the config for the CRJ is great but the A32NX still has a bunch of stuff that seems like it doesn't work or isn't programmed. Starting with the encoders, I'd like to configure the following items:

1: Barometric - Working
2: MFD Mode - Not working
3: MFD Range - I think this worked in the non-experimental version of the A32NX but does nothing now
4: Speed - Working
5: Heading - Working
6: Altitude - Working
7: V/S - Not working
8: Radios - Not working

Can anyone help? I'm not very good with coding/programming unfortunately.

**Comment by bigbrowncow** [comment]
This layout uses the template for the CRJ so encoders and buttons might not be where you are expecting.

Add to Community\mobiflight-event-module\modules\events.user.txt

```
A32nx_Neo_MFD_NAV_MODE_1_DEC#(L:A32NX_EFIS_L_ND_MODE) 0 > if{ (L:A32NX_EFIS_L_ND_MODE) 1 - (>L:A32NX_EFIS_L_ND_MODE) }
A32nx_Neo_MFD_NAV_MODE_1_INC#(L:A32NX_EFIS_L_ND_MODE) 4 < if{ (L:A32NX_EFIS_L_ND_MODE) 1 + (>L:A32NX_EFIS_L_ND_MODE) }
A32nx_Neo_MFD_Range_1_DEC#(L:A32NX_EFIS_L_ND_RANGE) 0 > if{ (L:A32NX_EFIS_L_ND_RANGE) 1 - (>L:A32NX_EFIS_L_ND_RANGE) }
A32nx_Neo_MFD_Range_1_INC#(L:A32NX_EFIS_L_ND_RANGE) 5 < if{ (L:A32NX_EFIS_L_ND_RANGE) 1 + (>L:A32NX_EFIS_L_ND_RANGE) }
A32NX_BARO_MODE_PUSH#(L:XMLVAR_Baro1_Mode) 2 < if{ 1 (L:XMLVAR_Baro1_Mode) - } els{ 1 } (>L:XMLVAR_Baro1_Mode)

```

then this is the config_a320.json I am using with ther current experimental fbw a32nx.

push knob => quick press
pull knob => long press

```{
  "$schema": "./config.schema.json",
  "version": {
    "major": 1,
    "minor": 1,
    "patch": 0
  },
  "encoders": [
    {
      "index": 2,
      "event_up": "KOHLSMAN_INC",
      "event_down": "KOHLSMAN_DEC",
      "event_short_press": { "event": "MobiFlight.A32NX_BARO_MODE_PUSH", "type": "manual", "value": 1 },
      "event_long_press": { "event": "MobiFlight.XMLVAR_Baro1_Mode_STD", "type": "manual", "value": 1 }

    },
    {
      "index": 3,
      "event_up": "AP_SPD_VAR_INC",
      "event_down": "AP_SPD_VAR_DEC",
      "event_short_press": { "event": "MobiFlight.A32NX.FCU_SPD_PUSH", "type": "manual", "value": 1 },
      "event_long_press": { "event": "MobiFlight.A32NX.FCU_SPD_PULL", "type": "manual", "value": 1 }
    },
    {
      "index": 5,
      "event_up": "HEADING_BUG_INC",
      "event_down": "HEADING_BUG_DEC",
      "event_short_press": { "event": "MobiFlight.A32NX.FCU_HDG_PUSH", "type": "manual", "value": 1 },
      "event_long_press": { "event": "MobiFlight.A32NX.FCU_HDG_PULL", "type": "manual", "value": 1 }
    },
    {
      "index": 7,
      "event_up": "AP_ALT_VAR_INC",
      "event_down": "AP_ALT_VAR_DEC",
      "event_short_press": { "event": "MobiFlight.A32NX.FCU_ALT_PUSH", "type": "manual", "value": 1 },
      "event_long_press": { "event": "MobiFlight.A32NX.FCU_ALT_PULL", "type": "manual", "value": 1 }
      
    },
    {
      "index": 8,
      "event_up": "AP_VS_VAR_INC",
      "event_down": "AP_VS_VAR_DEC",
      "event_short_press": { "event": "MobiFlight.A32NX.FCU_VS_PUSH", "type": "manual", "value": 1 },
      "event_long_press": { "event": "MobiFlight.A32NX.FCU_VS_PULL", "type": "manual", "value": 1 }
    },
    {
      "index": 6,
      "event_up": "NAV1_RADIO_WHOLE_INC",
      "event_down": "NAV1_RADIO_WHOLE_DEC",
      "alternate_event_up": "NAV1_RADIO_FRACT_INC",
      "alternate_event_down": "NAV1_RADIO_FRACT_DEC",
      "event_short_press": "{alternate}",
      "event_long_press": "NAV1_RADIO_SWAP"
    },
    {
      "index": 1,
      "event_up": { "event": "MobiFlight.A32nx_Neo_MFD_NAV_MODE_1_INC", "type": "manual", "value": 1 },
      "event_down": { "event": "MobiFlight.A32nx_Neo_MFD_NAV_MODE_1_DEC", "type": "manual", "value": 1 },
      "alternate_event_up": { "event": "MobiFlight.A32nx_Neo_MFD_Range_1_INC", "type": "manual", "value": 1 },
      "alternate_event_down": { "event": "MobiFlight.A32nx_Neo_MFD_Range_1_DEC", "type": "manual", "value": 1 },
      "event_press": "{alternate}"
	  
	  
    },
	   {
      "index": 4,

      "event_up": "COM_RADIO_WHOLE_INC",
      "event_down": "COM_RADIO_WHOLE_DEC",
      "alternate_event_up": "COM_RADIO_FRACT_INC",
      "alternate_event_down": "COM_RADIO_FRACT_DEC",
      "event_short_press": "{alternate}",
      "event_long_press": "COM_STBY_RADIO_SWAP"
     },
	 { 
      "index": 12,

      "event_up": "COM2_RADIO_WHOLE_INC",
      "event_down": "COM2_RADIO_WHOLE_DEC",
      "alternate_event_up": "COM2_RADIO_FRACT_INC",
      "alternate_event_down": "COM2_RADIO_FRACT_DEC",
      "event_short_press": "{alternate}",
      "event_long_press": "COM2_RADIO_SWAP"
     }
 
  ],
  "buttons": [
    {
      "index": 9,
      "event_press": [ "TOGGLE_FLIGHT_DIRECTOR", "AP_MASTER" ],
      "simvar_led": "AUTOPILOT_FLIGHT_DIRECTOR_ACTIVE"
    },
    {
      "index": 12,
      "event_press": "AP_BC_HOLD",
      "simvar_led": "AUTOPILOT_BACKCOURSE_HOLD"
    },
    {
      "index": 5,
      "event_press": "AP_HDG_HOLD",
      "mobiflightsimvar_led": "(L:A32NX_FCU_HDG_MANAGED_DOT)"
    },
    {
      "index": 7,
	        "mobiflightsimvar_led": "(L:A32NX_FCU_ALT_MANAGED)"

     },
    {
      "index": 8
    },
    {
      "index": 7
    },
    {
      "index": 15
    },
    {
      "index": 1,
      "event_press": "AP_MASTER",
      "mobiflightsimvar_led": "(L:A32NX_AUTOPILOT_1_ACTIVE)"
    },
    {
      "index": 6,
      "event_press": "AP_NAV1_HOLD",
      "simvar_led": "AUTOPILOT_NAV1_LOCK"
    },
    {
      "index": 4,
      "event_press": "AP_APR_HOLD",
      "mobiflightsimvar_led": "(L:A32NX_FCU_APPR_MODE_ACTIVE)"
    },
    {
      "index": 12
    },
    {
      "index": 3,
      "event_press": "FLIGHT_LEVEL_CHANGE",
      "mobiflightsimvar_led": "(L:A32NX_FCU_SPD_MANAGED_DOT)"
    },
    {
      "index": 14
    }
  ],
  "faders": [
    {
      "index": 1,
      "event_change": "AXIS_SPOILER_SET",
      "min_value": -16384,
      "max_value": 16384
    }
  ],
  "triggers": [
     {
      "trigger_type": "condition-file",
      "simvar": "AUTOPILOT_HEADING_LOCK_DIR",
      "condition": "Functions/Generic/hdg-lock-sync.jinja2"
    }
  ],
  "initialization": {
    "global_variables": [
      {
        "name": "HEADING_ENC_IDX",
        "value": 5
      },
	  {
        "name": "MFD_RANGE",
        "value": 10
      }
    ]
  }
}
```

**Reply by P1NBA11ER** [reply]
Thanks for helping. The mobiflight lines are already present in my events txt file and unfortunately, trying your config crashes x-touch for me. Loading it in atom it seems like there's a formatting error at some point since the code coloring is shifting, but I can't seem to find it.

I also tried pasting only your code for the MFD modes/range into the custom config I've been working on and it still didn't work.

EDIT: I think the crash was related to the jinja script which I don't have installed.

**Reply by bigbrowncow** [reply]
Yeah you could take out the jinja script as its not needed on the a320 - I didn’t get that far…the MFD RANGE variable isn’t needed either

**Reply by P1NBA11ER** [reply]
I've got it loading again but the MFD script still doesn't work at all. This script works for you with the experimental version of the A32NX? I've even tried switching liveries to the default ones just in case.

**Comment by P1NBA11ER** [comment]
For anyone looking to get the EFIS Mode and Range to work with the experimental version of the A32NX, I've finally sorted this out with some help from the MobiFlight community on Discord. 

Add the following to the MobiFlight events.user.txt:
// A32X Experimental EFIS Mode
A32NX_EFIS_L_ND_MODE_LS#0 (>L:A32NX_EFIS_L_ND_MODE)
A32NX_EFIS_L_ND_MODE_VOR#1 (>L:A32NX_EFIS_L_ND_MODE)
A32NX_EFIS_L_ND_MODE_NAV#2 (>L:A32NX_EFIS_L_ND_MODE)
A32NX_EFIS_L_ND_MODE_ARC#3 (>L:A32NX_EFIS_L_ND_MODE)
A32NX_EFIS_L_ND_MODE_PLAN#4 (>L:A32NX_EFIS_L_ND_MODE)
// A32X Experimental EFIS Range
A32NX_EFIS_L_ND_RANGE_10#0 (>L:A32NX_EFIS_L_ND_RANGE)
A32NX_EFIS_L_ND_RANGE_20#1 (>L:A32NX_EFIS_L_ND_RANGE)
A32NX_EFIS_L_ND_RANGE_40#2 (>L:A32NX_EFIS_L_ND_RANGE)
A32NX_EFIS_L_ND_RANGE_80#3 (>L:A32NX_EFIS_L_ND_RANGE)
A32NX_EFIS_L_ND_RANGE_160#4 (>L:A32NX_EFIS_L_ND_RANGE)
A32NX_EFIS_L_ND_RANGE_320#5 (>L:A32NX_EFIS_L_ND_RANGE)

A sample of how to alter the config_a320.json...

  "event": [ "{% if data.get_global_variable('MFD_NAV_MODE') == 0 %}",
                     "{{ data.trigger_event('MobiFlight.**A32NX_EFIS_L_ND_MODE**_VOR', 1) }}",
                     "{{ data.set_global_variable('MFD_NAV_MODE', 1) }}",
                   "{% elif data.get_global_variable('MFD_NAV_MODE') == 1 %}",
                     "{{ data.trigger_event('MobiFlight.A**32NX_EFIS_L_ND_MODE**_NAV', 1) }}",
                     "{{ data.set_global_variable('MFD_NAV_MODE', 2) }}",


The range is in a jinja file... change the variable here as well

{% if data.get_global_variable('MFD_RANGE') == 320 %}
    {{ data.trigger_event('MobiFlight.**A32NX_EFIS_L_ND_RANGE**_160', 1) }}
    {{ data.set_global_variable('MFD_RANGE', 160) }}
{% elif data.get_global_variable('MFD_RANGE') == 160 %}
    {{ data.trigger_event('MobiFlight.**A32NX_EFIS_L_ND_RANGE**_80', 1) }}
    {{ data.set_global_variable('MFD_RANGE', 80) }}

**Comment by maxverstappen** [comment]
As an programming noob, can you share your files?

**Reply by P1NBA11ER** [reply]
I don't see a way to add files and I'm not a programmer either, at all. This is pretty simple find and replace stuff. You just need to follow the example above and add the noted content to the events.user.txt file in the folder Community\mobiflight-event-module\modules (remove inactive from the end of the default files' name) then swap out the old variable names for the new. So change A320_Neo_MFD_Range_1 and A320_Neo_MFD_NAV_MODE_1 to A32NX_EFIS_L_ND_RANGE and A32NX_EFIS_L_ND_MODE

**Reply by maxverstappen** [reply]
You can add the three files to a zip file and drag the zipfile to the area where you write an answer.
[t.zip](https://github.com/maartentamboer/X-Touch-Mini-FS2020/files/7332409/t.zip)

**Reply by P1NBA11ER** [reply]
[Configurations.zip](https://github.com/maartentamboer/X-Touch-Mini-FS2020/files/7332437/Configurations.zip)
[events.user.txt](https://github.com/maartentamboer/X-Touch-Mini-FS2020/files/7332441/events.user.txt)

Ok, here's the files for MY config. There's probably some stuff in here that doesn't work. I'm still chipping away at organizing it in a way I like and which makes sense to me.

**Reply by maxverstappen** [reply]
thx
